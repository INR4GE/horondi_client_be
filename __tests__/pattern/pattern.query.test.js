const { gql } = require('@apollo/client');
/* eslint-disable no-undef */
const client = require('../../utils/apollo-test-client');
require('dotenv').config();
const { PATTERN_NOT_FOUND } = require('../../error-messages/pattern.messages');
const { adminLogin } = require('../helper-functions');

const {
  patternDoesNotExistId,
  skip,
  limit,
  wrongLimit,
  wrongSkip,
  user,
  languageTypeName,
  imageTypeName,
  queryPatternToAdd,
} = require('./pattern.variables');

let token;
let patternId;

describe('pattern query tests', () => {
  beforeAll(async () => {
    token = await adminLogin(user);
    const res = await client
      .mutate({
        mutation: gql`
          mutation($pattern: PatternInput!) {
            addPattern(pattern: $pattern) {
              ... on Pattern {
                _id
              }
              ... on Error {
                message
                statusCode
              }
            }
          }
        `,

        variables: { pattern: queryPatternToAdd },
        context: { headers: { token } },
      })
      .catch(e => e);
    patternId = res.data.addPattern._id;
  });

  describe('tests for all patterns and one pattern', () => {
    test('#1 Should receive all patterns', async () => {
      const res = await client
        .query({
          query: gql`
            query {
              getAllPatterns {
                items {
                  name {
                    lang
                    value
                  }
                  description {
                    lang
                    value
                  }
                  images {
                    large
                    medium
                    small
                    thumbnail
                  }
                  material
                  handmade
                  available
                }
              }
            }
          `,
        })
        .catch(e => e);

      expect(res.data.getAllPatterns).toMatchSnapshot();
      expect(res.data.getAllPatterns).toBeDefined();
      expect(res.data.getAllPatterns.items).toContainEqual({
        __typename: 'Pattern',
        name: queryPatternToAdd.name.map(item => ({
          ...languageTypeName,
          ...item,
        })),
        description: queryPatternToAdd.description.map(item => ({
          ...languageTypeName,
          ...item,
        })),
        images: { ...imageTypeName, ...queryPatternToAdd.images },
        material: queryPatternToAdd.material,
        handmade: false,
        available: true,
      });
    });
    test('#2 Should receive one pattern', async () => {
      const res = await client
        .query({
          query: gql`
            query($id: ID!) {
              getPatternById(id: $id) {
                ... on Pattern {
                  name {
                    lang
                    value
                  }
                  description {
                    lang
                    value
                  }
                  images {
                    large
                    medium
                    small
                    thumbnail
                  }
                  material
                  handmade
                  available
                }
              }
            }
          `,
          variables: { id: patternId },
        })
        .catch(e => e);

      expect(res.data.getPatternById).toMatchSnapshot();
      expect(res.data.getPatternById).toBeDefined();
      expect(res.data.getPatternById.name).toBeInstanceOf(Array);
      expect(res.data.getPatternById).toHaveProperty(
        'name',
        queryPatternToAdd.name.map(item => ({
          ...languageTypeName,
          ...item,
        }))
      );
      expect(res.data.getPatternById.images).toBeInstanceOf(Object);
      expect(res.data.getPatternById).toHaveProperty('images', {
        ...imageTypeName,
        ...queryPatternToAdd.images,
      });
      expect(res.data.getPatternById.images).toBeInstanceOf(Object);
      expect(res.data.getPatternById).toHaveProperty(
        'description',
        ...queryPatternToAdd.description.map(item => ({
          ...languageTypeName,
          ...item,
        }))
      );
      expect(res.data.getPatternById.description).toBeInstanceOf(Array);
      expect(res.data.getPatternById).toHaveProperty(
        'material',
        queryPatternToAdd.material
      );
      expect(res.data.getPatternById).toHaveProperty(
        'handmade',
        queryPatternToAdd.handmade
      );
      expect(res.data.getPatternById).toHaveProperty(
        'available',
        queryPatternToAdd.available
      );
    });
    test('#3 request not existing pattern should throw error', async () => {
      const res = await client
        .query({
          query: gql`
            query($id: ID!) {
              getPatternById(id: $id) {
                ... on Pattern {
                  name {
                    lang
                    value
                  }
                  description {
                    lang
                    value
                  }
                  images {
                    large
                    medium
                    small
                    thumbnail
                  }
                  material
                  handmade
                  available
                }
                ... on Error {
                  statusCode
                  message
                }
              }
            }
          `,
          variables: { id: patternDoesNotExistId },
        })
        .catch(e => e);

      expect(res.data.getPatternById).toBeDefined();
      expect(res.data.getPatternById).toHaveProperty('statusCode', 404);
      expect(res.data.getPatternById).toHaveProperty(
        'message',
        PATTERN_NOT_FOUND
      );
    });

    test('pattern pagination test', async () => {
      const res = await client
        .query({
          variables: { skip, limit },
          query: gql`
            query($skip: Int, $limit: Int) {
              getAllPatterns(skip: $skip, limit: $limit) {
                items {
                  name {
                    lang
                    value
                  }
                  description {
                    lang
                    value
                  }
                  images {
                    large
                    medium
                    small
                    thumbnail
                  }
                  material
                  handmade
                  available
                }
                count
              }
            }
          `,
        })
        .catch(e => e);

      expect(res.data.getAllPatterns).toMatchSnapshot();
      expect(res.data.getAllPatterns.items).toHaveLength(5);
      expect(res.data.getAllPatterns.count).toEqual(16);
    });
    test('pattern pagination test with wrong arguments', async () => {
      const res = await client
        .query({
          variables: { skip: wrongLimit, limit: wrongSkip },
          query: gql`
            query($skip: Int, $limit: Int) {
              getAllPatterns(skip: $skip, limit: $limit) {
                items {
                  name {
                    lang
                    value
                  }
                  description {
                    lang
                    value
                  }
                  images {
                    large
                    medium
                    small
                    thumbnail
                  }
                  material
                  handmade
                  available
                }
                count
              }
            }
          `,
        })
        .catch(e => e);
      expect(res.graphQLErrors[0].message).toEqual(
        'Skip value must be non-negative, but received: -5'
      );
    });
  });
  afterAll(async () => {
    await client
      .mutate({
        mutation: gql`
          mutation($id: ID!) {
            deletePattern(id: $id) {
              ... on Pattern {
                _id
              }
              ... on Error {
                statusCode
                message
              }
            }
          }
        `,
        variables: { id: patternId },
        context: { headers: { token } },
      })
      .catch(e => e);
    expect(res.graphQLErrors[0].message).toEqual(
      'Skip value must be non-negative, but received: -5'
    );
  });
});
