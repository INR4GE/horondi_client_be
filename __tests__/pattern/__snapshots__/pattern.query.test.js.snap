// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pattern queries tests for all patterns and one pattern #1 Should receive all patterns 1`] = `
Object {
  "items": Array [
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням рожевої хвильки",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with pink wave pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr4j5dkebkw5cy_pink-wave.jpg",
        "medium": "medium_335nr4j5dkebkw5cy_pink-wave.jpg",
        "small": "small_335nr4j5dkebkw5cy_pink-wave.jpg",
        "thumbnail": "thumbnail_335nr4j5dkebkw5cy_pink-wave.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Рожева хвилька",
        },
        Object {
          "lang": "en",
          "value": "Pink wave",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням синьої хвильки",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with blue wave pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr4j5dkebkvle7_blue-wave.jpg",
        "medium": "medium_335nr4j5dkebkvle7_blue-wave.jpg",
        "small": "small_335nr4j5dkebkvle7_blue-wave.jpg",
        "thumbnail": "thumbnail_335nr4j5dkebkvle7_blue-wave.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Синя хвилька",
        },
        Object {
          "lang": "en",
          "value": "Blue wave",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням синьо-рожевої хвильки",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with blue-pink wave pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr431gkebp19ht_blue-pink.jpg",
        "medium": "medium_335nr431gkebp19ht_blue-pink.jpg",
        "small": "small_335nr431gkebp19ht_blue-pink.jpg",
        "thumbnail": "thumbnail_335nr431gkebp19ht_blue-pink.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Синьо-рожева хвилька",
        },
        Object {
          "lang": "en",
          "value": "Blue-pink wave",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням стрілок",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with arrow pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr4j5dkebkv6hj_arrows.jpg",
        "medium": "medium_335nr4j5dkebkv6hj_arrows.jpg",
        "small": "small_335nr4j5dkebkv6hj_arrows.jpg",
        "thumbnail": "thumbnail_335nr4j5dkebkv6hj_arrows.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Стрілки",
        },
        Object {
          "lang": "en",
          "value": "Arrows",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із синім визерунком",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with blue pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr4j5dkebkvf6o_blue.jpg",
        "medium": "medium_335nr4j5dkebkvf6o_blue.jpg",
        "small": "small_335nr4j5dkebkvf6o_blue.jpg",
        "thumbnail": "thumbnail_335nr4j5dkebkvf6o_blue.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Синій",
        },
        Object {
          "lang": "en",
          "value": "Blue",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із чорним визерунком",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with black pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr4j5dkebkv6hr_black.jpg",
        "medium": "medium_335nr4j5dkebkv6hr_black.jpg",
        "small": "small_335nr4j5dkebkv6hr_black.jpg",
        "thumbnail": "thumbnail_335nr4j5dkebkv6hr_black.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Чорний",
        },
        Object {
          "lang": "en",
          "value": "Black",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із червоним визерунком",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with red pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr435lkebp5bxe_red.jpg",
        "medium": "medium_335nr435lkebp5bxe_red.jpg",
        "small": "small_335nr435lkebp5bxe_red.jpg",
        "thumbnail": "thumbnail_335nr435lkebp5bxe_red.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Червоний",
        },
        Object {
          "lang": "en",
          "value": "Red",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням вишивки",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with embroidery pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr435lkebp2r7d_embroidery.jpg",
        "medium": "medium_335nr435lkebp2r7d_embroidery.jpg",
        "small": "small_335nr435lkebp2r7d_embroidery.jpg",
        "thumbnail": "thumbnail_335nr435lkebp2r7d_embroidery.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Вишивка",
        },
        Object {
          "lang": "en",
          "value": "Embroidery",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням оленів",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with deer pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr4j5dkebkvleg_deers.jpg",
        "medium": "medium_335nr4j5dkebkvleg_deers.jpg",
        "small": "small_335nr4j5dkebkvleg_deers.jpg",
        "thumbnail": "thumbnail_335nr4j5dkebkvleg_deers.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Олені",
        },
        Object {
          "lang": "en",
          "value": "Deers",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням людей",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with people pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr4j5dkebkvzrt_people.jpg",
        "medium": "medium_335nr4j5dkebkvzrt_people.jpg",
        "small": "small_335nr4j5dkebkvzrt_people.jpg",
        "thumbnail": "thumbnail_335nr4j5dkebkvzrt_people.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Люди",
        },
        Object {
          "lang": "en",
          "value": "People",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням квітів",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with flower pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr4j5dkebkvus7_flowers.jpg",
        "medium": "medium_335nr4j5dkebkvus7_flowers.jpg",
        "small": "small_335nr4j5dkebkvus7_flowers.jpg",
        "thumbnail": "thumbnail_335nr4j5dkebkvus7_flowers.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Квіти",
        },
        Object {
          "lang": "en",
          "value": "Flowers",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням сірих ромбів",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with grey diamond pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr435lkebp4pbv_grey-diamonds.jpg",
        "medium": "medium_335nr435lkebp4pbv_grey-diamonds.jpg",
        "small": "small_335nr435lkebp4pbv_grey-diamonds.jpg",
        "thumbnail": "thumbnail_335nr435lkebp4pbv_grey-diamonds.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Сірі ромби",
        },
        Object {
          "lang": "en",
          "value": "Grey diamonds",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням зелених ромбів",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with green diamond pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr435lkebp3yc5_green-diamonds.jpg",
        "medium": "medium_335nr435lkebp3yc5_green-diamonds.jpg",
        "small": "small_335nr435lkebp3yc5_green-diamonds.jpg",
        "thumbnail": "thumbnail_335nr435lkebp3yc5_green-diamonds.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Зелені ромби",
        },
        Object {
          "lang": "en",
          "value": "Green diamonds",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням червоних ромбів",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with red diamond pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_pljirbf0kfmetg5l_158.jpg",
        "medium": "medium_pljirbf0kfmetg5l_158.jpg",
        "small": "small_pljirbf0kfmetg5l_158.jpg",
        "thumbnail": "thumbnail_pljirbf0kfmetg5l_158.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Червоні ромби",
        },
        Object {
          "lang": "en",
          "value": "Red diamonds",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням синіх ромбів",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with blue diamond pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr42zgkeboz0w5_blue-diamonds.jpg",
        "medium": "medium_335nr42zgkeboz0w5_blue-diamonds.jpg",
        "small": "small_335nr42zgkeboz0w5_blue-diamonds.jpg",
        "thumbnail": "thumbnail_335nr42zgkeboz0w5_blue-diamonds.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Сині ромби",
        },
        Object {
          "lang": "en",
          "value": "Blue diamonds",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням синьої хвильки",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with blue wave pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr4j5dkebkvle7_blue-wave.jpg",
        "medium": "medium_335nr4j5dkebkvle7_blue-wave.jpg",
        "small": "small_335nr4j5dkebkvle7_blue-wave.jpg",
        "thumbnail": "thumbnail_335nr4j5dkebkvle7_blue-wave.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Синя хвилька",
        },
        Object {
          "lang": "en",
          "value": "Blue wave",
        },
      ],
    },
    Object {
      "available": false,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "dasdas",
        },
        Object {
          "lang": "en",
          "value": "dasd",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_99dpjeyckg16c1ix_gream.jpg",
        "medium": "medium_99dpjeyckg16c1ix_gream.jpg",
        "small": "small_99dpjeyckg16c1ix_gream.jpg",
        "thumbnail": "thumbnail_99dpjeyckg16c1ix_gream.jpg",
      },
      "material": "dasdas",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "dasdasd",
        },
        Object {
          "lang": "en",
          "value": "dasdas",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "test value",
        },
        Object {
          "lang": "en",
          "value": "test value",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_0_test-file",
        "medium": "medium_0_test-file",
        "small": "small_0_test-file",
        "thumbnail": "thumbnail_0_test-file",
      },
      "material": "test",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "test value",
        },
        Object {
          "lang": "en",
          "value": "test value",
        },
      ],
    },
  ],
}
`;

exports[`Pattern queries tests for all patterns and one pattern #2 Should receive one pattern 1`] = `
Object {
  "available": true,
  "description": Array [
    Object {
      "lang": "uk",
      "value": "test value",
    },
    Object {
      "lang": "en",
      "value": "test value",
    },
  ],
  "handmade": false,
  "images": Object {
    "large": "large_0_test-file",
    "medium": "medium_0_test-file",
    "small": "small_0_test-file",
    "thumbnail": "thumbnail_0_test-file",
  },
  "material": "test",
  "name": Array [
    Object {
      "lang": "uk",
      "value": "test value",
    },
    Object {
      "lang": "en",
      "value": "test value",
    },
  ],
}
`;

exports[`Pattern queries tests for all patterns and one pattern pattern pagination test 1`] = `
Object {
  "count": 18,
  "items": Array [
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням рожевої хвильки",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with pink wave pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr4j5dkebkw5cy_pink-wave.jpg",
        "medium": "medium_335nr4j5dkebkw5cy_pink-wave.jpg",
        "small": "small_335nr4j5dkebkw5cy_pink-wave.jpg",
        "thumbnail": "thumbnail_335nr4j5dkebkw5cy_pink-wave.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Рожева хвилька",
        },
        Object {
          "lang": "en",
          "value": "Pink wave",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням синьої хвильки",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with blue wave pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr4j5dkebkvle7_blue-wave.jpg",
        "medium": "medium_335nr4j5dkebkvle7_blue-wave.jpg",
        "small": "small_335nr4j5dkebkvle7_blue-wave.jpg",
        "thumbnail": "thumbnail_335nr4j5dkebkvle7_blue-wave.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Синя хвилька",
        },
        Object {
          "lang": "en",
          "value": "Blue wave",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням синьо-рожевої хвильки",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with blue-pink wave pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr431gkebp19ht_blue-pink.jpg",
        "medium": "medium_335nr431gkebp19ht_blue-pink.jpg",
        "small": "small_335nr431gkebp19ht_blue-pink.jpg",
        "thumbnail": "thumbnail_335nr431gkebp19ht_blue-pink.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Синьо-рожева хвилька",
        },
        Object {
          "lang": "en",
          "value": "Blue-pink wave",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із зображенням стрілок",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with arrow pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr4j5dkebkv6hj_arrows.jpg",
        "medium": "medium_335nr4j5dkebkv6hj_arrows.jpg",
        "small": "small_335nr4j5dkebkv6hj_arrows.jpg",
        "thumbnail": "thumbnail_335nr4j5dkebkv6hj_arrows.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Стрілки",
        },
        Object {
          "lang": "en",
          "value": "Arrows",
        },
      ],
    },
    Object {
      "available": true,
      "description": Array [
        Object {
          "lang": "uk",
          "value": "Фабричний гобелен із синім визерунком",
        },
        Object {
          "lang": "en",
          "value": "Manufactured tapestry with blue pattern",
        },
      ],
      "handmade": false,
      "images": Object {
        "large": "large_335nr4j5dkebkvf6o_blue.jpg",
        "medium": "medium_335nr4j5dkebkvf6o_blue.jpg",
        "small": "small_335nr4j5dkebkvf6o_blue.jpg",
        "thumbnail": "thumbnail_335nr4j5dkebkvf6o_blue.jpg",
      },
      "material": "Cotton",
      "name": Array [
        Object {
          "lang": "uk",
          "value": "Синій",
        },
        Object {
          "lang": "en",
          "value": "Blue",
        },
      ],
    },
  ],
}
`;
